api/v1beta1/upgradetaskphase_types.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+type UpgradeTaskPhase string
+
+const (
```

Comment:
same as above comment, we need a setting in spec to tell which target state we want to drive to

Reply:
`TargetPhase` is added to address this comment. To keep it simple, the same type is reused.